### ISSUE TITLE
AZURE COMMUNICATION SERVICE CALL FEATURE (.NET CORE, C# SDK)

### ISSUE THREAD

**Issue by AnsariNaushad** [Azure/Communication#72]
Microsoft has provided .Net SDKs for chatting and sending SMS using Azure communication services.
We want to implement call feature in our application, as per the documentation call is not available in .Net SDK.
https://github.com/Azure/communication

When we can expect .net SDK for call, or Microsoft have provided any alternate solution?
Thanks

**Comment by RinaRish** [comment]
Hi @AnsariNaushad! Thanks for your feedback. So .NET SDK is on the roadmap but we can't commit to any timeline, unfortunately. Regarding your second question could you please give us more details about your specific scenario and we can see if there is an alternative.

**Comment by AnsariNaushad** [comment]
Thanks Rina for quick response.
I got sample application from your Git repository. It has calling (samples\Group Calling Hero Sample\Web\Calling) feature but application is build on node with redux.
I want to integrate same features in my .net core application using C#, Java Script.

Second question is like can I utilize these Java Scripts for my .net application.

**Comment by akania** [comment]
@AnsariNaushad can you please clarify what's the goal, what that .net application should do? enable users to make/receive calls? would it utilize UWP - that would be one potential solution.

**Comment by AnsariNaushad** [comment]
Yes Artur, I want to  enable users to make/receive calls.
That can be audio or video.
What is UWP kindly let me know, and how it is going to help me to do the
same.

**Comment by tophpalmer** [comment]
hey @AnsariNaushad , we want to make sure that we're supporting the exact Windows platform you need. You can see a [list of platforms here](https://docs.microsoft.com/en-us/windows/apps/desktop/). The choices are UWP, C++/Win32, WPF, or Windows Forms.

**Comment by FlapperA** [comment]
Any progress on this? I am also waiting on .Net SDK.  I want to use with .Net Core and ASP.Net

**Comment by mariusu-msft** [comment]
Hi @FlapperA,

Currently we are planning a client-side Windows SDK that will be available for the following app platforms: `UWP, C++/Win32, WPF, or Windows Forms.`

If you require a Windows server SDK to be used with ASP.NET Core or such, can you provide us with some scenarios you are looking at? We will direct the feedback to the team looking at Server scenarios.

Thanks,
Marius

**Comment by FlapperA** [comment]
The reason for using .Net Core (ASP.Net) over Angular or ReactJS is maintainability. Recent experiences with Angular (and a bit ReactJS) has learned that very rigorous updating of the platform is needed to not expose any security leaks. 

As we intend to use ACS in a scenario where security is of the utmost importance, we want to stay away from platforms with an unpredictable roadmap. .Net Core is much easier to support and we noticed much less security issues.

**Comment by mariusu-msft** [comment]
@FlapperA do you mind sharing a server scenario or API you would like to use from ASP.Net Core?

**Comment by FlapperA** [comment]
@mariusu-msft That is possible, but preferably in a private message as this might be company sensitive.

**Comment by mariusu-msft** [comment]
@FlapperA sounds good, please sends us an email at `acsfeedback @ microsoft.com`

**Comment by ProbablePrime** [comment]
I've removed the author feedback label as we have multiple reports and requests for this item.

**Comment by FlapperA** [comment]
> @FlapperA sounds good, please sends us an email at `acsfeedback @ microsoft.com`

Hi Marius,

I tried several times to send you an email to that address, but it looks like it bounces all the time.

**Comment by jwittner** [comment]
Embedded video calls, akin to Remote Assist, is a frequent ask from companies building HoloLens applications and not having UWP support from Azure Communication Services has been a bit of a disappointment in that space. Adding UWP support would unlock tons of value on that device.

**Comment by LoadLibrary** [comment]
@jwittner , the ACS Calling SDK does support UWP. Feel free to add https://www.nuget.org/packages/Azure.Communication.Calling as dependency of your UWP project. We also have a quickstart page that will help you with the integration https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/voice-video-calling/getting-started-with-calling?pivots=platform-windows .

**Comment by KevinKennedy** [comment]
I can run the calling sample above fine as a desktop app on x64.  If I build it for arm64 and deploy it to a HoloLens I get the exception below when I make the call.  Specifically when calling StartCallAsync.  Is this expected to work on HoloLens?

System.Runtime.InteropServices.MissingInteropDataException
HResult=0x80131500
Message=System.Collections.Generic.IEnumerable`1[Azure.WinRT.Communication.ICommunicationIdentifier] is missing interop type marshalling data. To enable interop type marshalling data, add a MarshalObject directive to the application rd.xml file. For more information, please visit http://go.microsoft.com/fwlink/?LinkID=393965

**Comment by LoadLibrary** [comment]
@KevinKennedy The error message suggests Azure.WinRT.Communication.ICommunicationIdentifier is missing on the ARM64 package. Since issue #72 is tracking customers looking for a non-packaged drop of the ACS Calling SDK, can you open a new issue reporting your finding?
